shared/Dockerfile.shared

```diff
@@ -0,0 +1,25 @@
+FROM golang:1.18 as builder
```

Comment:
I dont think this Dockerfile is necessary. You can reuse the prysm one while changing the entrypoint for me the build-shared-states service.

Reply:
good catch, changed to use `Dockerfile.prysm`